### PR TITLE
Ft test

### PR DESCRIPTION
**Pull request by neymarjimoh**
**What does this PR do?**
- Writes tests of the product investo API endpoints
 
Summarise the main tasks handled in the pull request

**description of Task to be completed?**
- CRUD Users Tests
- Auth Middleware test

**How should this be manually tested?**
- Clone this repo, install dependencies `npm install`
- Checkout into this branch `git checkout ft-test`
- copy .env-example to .env `copy .env-example .env`
- run `npm run test`  to test

Give bullet point instructions on how to setup the code and test it manually (Assume no prior experience on JS or code)

**Any background context you want to provide?**

Any pertinent information that should be considered

**What are the relevant PT Story ID?**
- None

[#173227224](https://www.pivotaltracker.com/n/projects/2451878)

**Screenshots (if appropriate)**
test error
test error

You can have screenshots where applicable

**Questions:**
- how do i run CRUD tests without the auth middleware ruuning, i tried stubbing it but it still runs when testing, if there's a way to just get the the token and store in a variable??
- 
If something is unclear or you want some questions to be addressed by your peers, mention them here

### CODE REVIEW

File: test/api/users.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+const { expect } = chai;
+chai.use(chaiHttp);
+const { User } = require('../../models');
+const sinon = require('sinon');
+const authMiddleware = require('../../middlewares/checkAuth');
+let app;
+var Token = '';
+
+describe('## USERS', function () {
+    let authToken;
```
Comment (Nicanor008) [inline test/api/users.js:13]:
```suggestion
    let authToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6ImpvaG5vZUBleGFtcGxlLmNvbSIsInVzZXJJZCI6IjVmMDQ0NTFlODBmMGJhMzk5ZGYxZjI2ZCIsInJvbGUiOiJ1c2VyIiwiaWF0IjoxNTk0MTE1Nzg4LCJleHAiOjE1OTQyMDIxODh9.VoSPs24TntksEk3fP-vV5VLYxmTOMHUa10SmGUt52DU;
 ```

initialize this variable

File: test/api/users.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    //     })
+    // });
+    
+
+    describe('#Get Users() ====', function () {
+        
+        it('should get all users from the database', function (done) {
+            chai.request(app)
+            .get('/api/v1/users')
+            // .set("Authorization", `Bearer ${authToken}`)
```
Comment (Nicanor008) [inline test/api/users.js:56]: Enable the authorization, since it requires auth

Comment (Nicanor008) [reply test/api/users.js:56]: This test should be successful, but it doesn't mean the endpoint doesn't work. Can you try it on postman

File: test/api/users.js (hunk 1 above)
Comment (neymarjimoh) [reply test/api/users.js:13]: ok

File: test/api/users.js (hunk 2 above)
Comment (neymarjimoh) [reply test/api/users.js:56]: the test still failed and it had a server error on postman

**Comment by neymarjimoh** [comment]
The test still doesn't work with the token but it works on postman

**Comment by Nicanor008** [comment]
> The test still doesn't work with the token but it works on postman

Let me check on it tomorrow.

Hint: Ensure to create an upload folder.

**Comment by Nicanor008** [comment]
You can work on the other task for the time being @neymarjimoh

**Comment by neymarjimoh** [comment]
Ok

**Comment by neymarjimoh** [comment]
ok, i will

**Comment by neymarjimoh** [comment]
Ok. I'd check this out @Nicanor008

**Comment by neymarjimoh** [comment]
it still did not work @Nicanor008

**Comment by neymarjimoh** [comment]
I've made the fix based on your review. It's now perfectly working

**Pull request merged**